Origin/feature/#42 dice shake sensor

Chfaeser opened the pull request:
Closes #42

Chfaeser committed:
Implemented basic shake-sensor functionality

Implemented basic functionality to roll the dice by shaking the device

Chfaeser committed:
Commit to save (not final)

This commit is only temporary

Chfaeser committed:
Added UI Element and code cleanup

Added text to tell the player to shake the device to roll the dice
Removed spaces, comments

Chfaeser committed:
Removed code smell

Chfaeser committed:
Merge develop into featurebranch

Resolved conflicts

Chfaeser committed:
Implementing an alternative to shaking the device

Implemented an alternative to shaking the device to roll
Changed some UI Elements
Player can now roll the dice by shaking the device or pressing the "Roll Dice" button after starting their turn

bferijan commented on app/src/main/java/com/ss20/se2/monopoly/view/GameboardActivity.java:
```diff
@@ -137,42 +152,113 @@ protected void onCreate(Bundle savedInstanceState) {
+						accel = accel * 0.9f + delta;
+
+						if(accel > 11){
+
+							mSensorManager.unregisterListener(this);
+							altbutton.setVisibility(View.GONE);
+
+							roll1 = dice.roll();
+							roll2 = dice2.roll();
+							amount = roll1 + roll2;
+
+							checkDouble(roll1, roll2);
+							if (checkDouble(roll1, roll2)){
+								doublescounter++;
+							}else{
+								doublescounter = 0;
+							}
+							if (doublescounter == 3){
+								//Move to jail
+							}
+							setOldBalance(currentPlayer.getBalance());
+							gameboard.move("Player 1", amount);
+
+							updateBalance.setText("");
+							currentPlayer.setCurrentPosition(gameboard.getPosition("Player 1"));
+							checkPlayersPosition(currentPlayer);
+
+							view_numberDice.setText("Roll 1: " + Integer.toString(roll1));
+							view_numberDice2.setText("Roll 2: " + Integer.toString(roll2));
+							view_position.setText(Integer.toString(gameboard.getPosition("Player 1")));
+							findViewById(R.id.playericon).setX(fields[gameboard.getPosition("Player 1")].getX());
+							findViewById(R.id.playericon).setY(fields[gameboard.getPosition("Player 1")].getY());
```
This should be extracted in to a method (e.g. processRoll) to prevent problems with merging.

bferijan commented on app/src/main/java/com/ss20/se2/monopoly/view/GameboardActivity.java:
```diff
@@ -137,42 +152,113 @@ protected void onCreate(Bundle savedInstanceState) {
+				altbutton.setOnClickListener(new View.OnClickListener(){
+					@Override
+					public void onClick(View v){
+
+						mSensorManager.unregisterListener(mSensorListener);
+						altbutton.setVisibility(View.GONE);
+
+						roll1 = dice.roll();
+						roll2 = dice2.roll();
+						amount = roll1 + roll2;
+
+						checkDouble(roll1, roll2);
+						if (checkDouble(roll1, roll2)){
+							doublescounter++;
+						}else{
+							doublescounter = 0;
+						}
+						if (doublescounter == 3){
+							//Move to jail
+						}
+						setOldBalance(currentPlayer.getBalance());
+						gameboard.move("Player 1", amount);
+
+						updateBalance.setText("");
+						currentPlayer.setCurrentPosition(gameboard.getPosition("Player 1"));
+						checkPlayersPosition(currentPlayer);
+
+						view_numberDice.setText("Roll 1: " + Integer.toString(roll1));
+						view_numberDice2.setText("Roll 2: " + Integer.toString(roll2));
+						view_position.setText(Integer.toString(gameboard.getPosition("Player 1")));
+						findViewById(R.id.playericon).setX(fields[gameboard.getPosition("Player 1")].getX());
+						findViewById(R.id.playericon).setY(fields[gameboard.getPosition("Player 1")].getY());
```
duplicate from previous review comment. (Please also insert method also here)

bferijan reviewed (approved):
Please have a look at the two comments. Functionallity works fine!

Chfaeser committed:
Merging develop into feature branch

Merging branches and smaller UI changes
Moved the duplication into the processRoll() method
